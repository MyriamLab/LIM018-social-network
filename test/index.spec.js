import {
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  signInWithPopup,
  sendEmailVerification,
} from '../src/firebase/config.js';

import {
  userRegister,
  userLogin,
  googleInicioSesion,
  sendEmailVerificationUser,
} from '../src/firebase/funcionesAuth.js';

jest.mock('../src/firebase/config.js'); //  mock habilitado para cargar el config falso no el original
// eslint-disable-next-line no-console

describe('register correo', () => {
  it('debería ser una función', () => {
    expect(typeof userRegister).toBe('function');
  });

  it('Debería poder registrar a un usuario', () => userRegister('frontend.la', '123456')
    .then(() => {
      // console.log(createUserWithEmailAndPassword.mock.calls);
      expect(createUserWithEmailAndPassword.mock.calls[0][1]).toBe('frontend.la');
      expect(createUserWithEmailAndPassword.mock.calls[0][2]).toBe('123456');
    }));
});

describe('Iniciar sesión', () => {
  it('debería ser una función', () => {
    expect(typeof userLogin).toBe('function');
  });

  it('Debería poder iniciar sesión', () => userLogin('frontend.la', '123456')
    .then(() => {
      // expect(createUserWithEmailAndPassword.mock.calls[0][0]).toBe('uid1');
      expect(signInWithEmailAndPassword.mock.calls[0][1]).toBe('frontend.la');
      expect(signInWithEmailAndPassword.mock.calls[0][2]).toBe('123456');
    }));
});

describe('Iniciar sesión con google', () => {
  it('debería ser una función', () => {
    expect(typeof googleInicioSesion).toBe('function');
  });

  it('Debería poder iniciar sesión con google', () => googleInicioSesion('frontend.la')
    .then(() => {
      // expect(createUserWithEmailAndPassword.mock.calls[0][0]).toBe('uid1');
      expect(signInWithPopup.mock.calls[0][1]).toBe('frontend.la');
    }));
});

// describe('Envía correo de verificación', () => {
//   it('debería ser una función', () => {
//     expect(typeof sendEmailVerificationUser).toBe('function');
//   });

//   it('Debería poder enviar un correo de verificación', () => sendEmailVerificationUser()
//     .then(() => {
//       console.log(sendEmailVerification.mock.calls);
//       expect(signInWithPopup.mock.calls[0]).toBe();
//     }));
// });
