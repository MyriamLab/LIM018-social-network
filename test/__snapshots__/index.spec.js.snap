// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register correo Debería de coincidir con el template 1`] = `
<section
  class="divForm"
>
  
  

  <div
    class="flex-direction divContainerImgForm"
  >
    
  
  
    <div
      class="flex-direction center divImgLogin"
    >
      
    
      <img
        alt="imágen de perritos y redes sociales"
        src="./imagenes/portada-dogs.png "
      />
      
  
    </div>
    
  
    <!--DIV DERECHO-->
    
  
    <div
      class="row-center divFormLogin"
    >
      
  
      <figure
        class="center padd-05"
      >
        
  
        <img
          alt="logo"
          src="./imagenes/logo-pets.png"
        />
        

      </figure>
      

    
      <!--inicio form register-->
      
    
      <div
        class="form-container"
      >
        
      
      
        <h3
          class="center"
        >
          Regístrate
        </h3>
        
      
        <form
          class="formLogin"
          id="formRegister"
        >
          
        
          <label
            class="group m-label"
          >
            Nombre Completo
          </label>
          
        
          <input
            id="idNameRegister"
            placeholder="Ej: María"
            required=""
            type="text"
          />
          

        
          <label
            class="group m-label"
          >
            E-mail
          </label>
          

        
          <input
            id="idEmailRegister"
            placeholder="pets@gmail.com"
            required=""
            type="email"
          />
          

        
          <label
            class="group m-label"
          >
            Contraseña 
          </label>
          

        
          <input
            id="idPasswordRegister"
            placeholder="Mínimo de 8 caracteres"
            required=""
            type="password"
          />
          

        
          <div
            class="buttonClass center"
          >
            
          
            <button
              id="idButtonRegister"
              type="submit"
            >
              Registrarse
            </button>
            
        
          </div>
          
        
          <dialog
            class="row-center"
            id="modalPadre"
          />
          
        
          <div
            class="flex-direction space-around padd-05"
          >
            
        
            <p>
              ¿Ya tienes una cuenta?
            </p>
            
        
            <a
              href="#/login"
            >
              
         Inicia Sesión
        
            </a>
            
      
          </div>
          
      
        </form>
        
    
      </div>
      
  
    </div>
    

  </div>
  
  
</section>
`;
